### PR TITLE
Run connectedCheck on CI

### PR DESCRIPTION
**Pull request by veyndan**
`TreehouseWidgetViewTest` doesn't compile. CI should've caught that.

### CODE REVIEW

**Comment by JakeWharton** [comment]
This does not actually run the test. So while you've forced it to compile, at least, there's no emulators running to actually execute the test.

I suspect we want a new CI job that runs an emulator and `connectedCheck` without enabling the samples to covers device tests in the core library.

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -35,6 +35,23 @@ jobs:
           path: build/dokka/htmlMultiModule/
           if-no-files-found: error
 
+  connected:
+    runs-on: ubuntu-latest
```
Comment (JakeWharton) [inline .github/workflows/build.yaml]:
```suggestion
    runs-on: macos-latest
```
The emulator is not accelerated on Linux unless you pay for it. Mac is our only option. It at least has some free minutes until we start paying.

**Review by JakeWharton — APPROVED** [review]

File: .github/workflows/build.yaml (hunk 1 above)
Comment (veyndan) [reply .github/workflows/build.yaml]: Ah gotcha, that explains why the CI build was taking forever. Thanks for all the help!

**Comment by veyndan** [comment]
So apparently auto-merge will merge regardless of the new CI job status 😬 

Any ideas on why the `connected` job can't seem to find paparazzi dependencies?

```
> Task :redwood-layout-view:mergeProjectDexDebugAndroidTest
e: file:///Users/runner/work/redwood/redwood/redwood-layout-composeui/src/androidInstrumentedTest/kotlin/app/cash/redwood/layout/composeui/ComposeUiFlexContainerTest.kt:25:17 Unresolved reference: paparazzi
e: file:///Users/runner/work/redwood/redwood/redwood-layout-composeui/src/androidInstrumentedTest/kotlin/app/cash/redwood/layout/composeui/ComposeUiFlexContainerTest.kt:26:17 Unresolved reference: paparazzi
e: file:///Users/runner/work/redwood/redwood/redwood-layout-composeui/src/androidInstrumentedTest/kotlin/app/cash/redwood/layout/composeui/ComposeUiFlexContainerTest.kt:44:19 Unresolved reference: Paparazzi
e: file:///Users/runner/work/redwood/redwood/redwood-layout-composeui/src/androidInstrumentedTest/kotlin/app/cash/redwood/layout/composeui/ComposeUiFlexContainerTest.kt:45:20 Unresolved reference: DeviceConfig
e: file:///Users/runner/work/redwood/redwood/redwood-layout-composeui/src/androidInstrumentedTest/kotlin/app/cash/redwood/layout/composeui/ComposeUiFlexContainerTest.kt:64:17 @Composable invocations can only happen from the context of a @Composable function
e: file:///Users/runner/work/redwood/redwood/redwood-layout-composeui/src/androidInstrumentedTest/kotlin/app/cash/redwood/layout/composeui/ComposeUiSpacerTest.kt:22:17 Unresolved reference: paparazzi
e: file:///Users/runner/work/redwood/redwood/redwood-layout-composeui/src/androidInstrumentedTest/kotlin/app/cash/redwood/layout/composeui/ComposeUiSpacerTest.kt:23:17 Unresolved reference: paparazzi
e: file:///Users/runner/work/redwood/redwood/redwood-layout-composeui/src/androidInstrumentedTest/kotlin/app/cash/redwood/layout/composeui/ComposeUiSpacerTest.kt:28:8 Unresolved reference: com
e: file:///Users/runner/work/redwood/redwood/redwood-layout-composeui/src/androidInstrumentedTest/kotlin/app/cash/redwood/layout/composeui/ComposeUiSpacerTest.kt:37:19 Unresolved reference: Paparazzi
e: file:///Users/runner/work/redwood/redwood/redwood-layout-composeui/src/androidInstrumentedTest/kotlin/app/cash/redwood/layout/composeui/ComposeUiSpacerTest.kt:38:20 Unresolved reference: DeviceConfig
e: file:///Users/runner/work/redwood/redwood/redwood-layout-composeui/src/androidInstrumentedTest/kotlin/app/cash/redwood/layout/composeui/ComposeUiSpacerTest.kt:40:21 Unresolved reference: SessionParams
```

**Comment by JakeWharton** [comment]
Your job needs added to the required list in the branch protection settings of the GitHub project if you want to block PRs on it.

Paparazzi is a JVM library and should not be used in instrumentation tests. Something is wrong with the dependency or folder structure of the tests in that module.

**Comment by veyndan** [comment]
> Your job needs added to the required list in the branch protection settings of the GitHub project if you want to block PRs on it.

That makes sense. I don't have permissions to access settings though. Would it be possible to have it added?

> Paparazzi is a JVM library and should not be used in instrumentation tests. Something is wrong with the dependency or folder structure of the tests in that module.

That was it! Fixed in https://github.com/cashapp/redwood/pull/878.

**Comment by JakeWharton** [comment]
Yep, once the fix lands I'll add it. Thanks for following up.

**Comment by JakeWharton** [comment]
Done

**Pull request merged**